test/typecheck/protolist-vs-array.boa

```diff
@@ -0,0 +1,6 @@
+o: output collection of int;
+repos: array of CodeRepository = input.code_repositories;
+id := function(fields: array of CodeRepository){
+  o << 1;
+};
+id(repos);
```

Comment:
You should call `id()` with `input.code_repositories` otherwise there is no reason it would fail, even with the old compiler.